From cf10960a92781e1263f274e2b30d46a9703a7dd5 Mon Sep 17 00:00:00 2001
From: Mohammad Rafi Shaik <quic_mohs@quicinc.com>
Date: Sat, 8 Mar 2025 08:40:53 +0530
Subject: [PATCH 7/8] PENDING: ASoC: qcom: qcs9100: Set dai format for MI2S
 interface

Set dai format for MI2S interface and update the
driver name as sa8775p which is base SoC for
qcs9100 and qcs9075 boards.

Change-Id: Ie5cc65c9f25ecff7ec92bdf9c2947e3d6bb56803
Signed-off-by: Mohammad Rafi Shaik <quic_mohs@quicinc.com>
Upstream-Status: Pending
---
 sound/soc/qcom/qcs9100.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/sound/soc/qcom/qcs9100.c b/sound/soc/qcom/qcs9100.c
index 631af7d3bdb2..681cf52a795a 100644
--- a/sound/soc/qcom/qcs9100.c
+++ b/sound/soc/qcom/qcs9100.c
@@ -83,6 +83,24 @@ static int qcs9100_snd_init(struct snd_soc_pcm_runtime *rtd)
 	return ret;
 }
 
+static int qcs9100_snd_startup(struct snd_pcm_substream *substream)
+{
+	struct snd_soc_pcm_runtime *rtd = snd_soc_substream_to_rtd(substream);
+	struct snd_soc_dai *cpu_dai = snd_soc_rtd_to_cpu(rtd, 0);
+	unsigned int fmt = SND_SOC_DAIFMT_BP_FP;
+
+	switch (cpu_dai->id) {
+	case PRIMARY_MI2S_RX ... QUATERNARY_MI2S_TX:
+	case PRIMARY_SDR_MI2S_RX ... QUINARY_SDR_MI2S_TX:
+		snd_soc_dai_set_fmt(cpu_dai, fmt);
+		break;
+	default:
+		break;
+	}
+
+	return 0;
+}
+
 static int qcs9100_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 				      struct snd_pcm_hw_params *params)
 {
@@ -182,6 +200,7 @@ static const struct snd_soc_dapm_route qcs9075_dapm_routes[] = {
 };
 
 static const struct snd_soc_ops qcs9100_be_ops = {
+	.startup = qcs9100_snd_startup,
 	.hw_params = qcs9100_snd_hw_params,
 	.hw_free = qcs9100_snd_hw_free,
 	.prepare = qcs9100_snd_prepare,
-- 
2.34.1

